### PR TITLE
Upgrade AppAuth-iOS to 1.0.0

### LINKED ISSUE #337: Update to use AppAuth 1.0.0
It looks like AppAuth 1.0.0 came out of beta a couple of weeks ago
https://github.com/openid/AppAuth-iOS/releases

### LINKED ISSUE #339: Add new pod installation instructions
This PR added a Podspec to the project https://github.com/FormidableLabs/react-native-app-auth/pull/313 

This changes the general installation instructions for projects created with `react-native init` (pre-RN0.60) The AppAuth dependency will no longer be added manually, it will be read from the Podspec after `react-native link`, but will also mean [additional editing](https://facebook.github.io/react-native/docs/integration-with-existing-apps#configuring-cocoapods-dependencies) for the Podspec.

### PR DESCRIPTION
**Pull request by kadikraman**
Fixes #337
Fixes #339

## Description

Upgrades the native AppAuth-iOS dependency.

**Note:** this will have to be a major release, due to a typo fix in a method name

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -241,9 +241,9 @@ AppAuth supports three options for dependency management.
    With [CocoaPods](https://guides.cocoapods.org/using/getting-started.html), add the following line to
    your `Podfile`:
 
-       pod 'AppAuth', '>= 0.94'
+       pod 'AppAuth', '1.0'
 
-   Then run `pod install`. Note that version 0.94 is the first of the library to support iOS 11.
+   Then run `pod install`.
```
Comment (thymikee) [inline README.md]: FYI, all instructions about linking on both paltforms are obsolete in RN 0.60 if the library works with autolinking. All you need to document now is how to use with Carthage or anything that is not CocoaPods or manual link

Comment (kadikraman) [reply README.md]: You're right, I'll update the instructions. Thanks for all your help on this!

**Comment by kadikraman** [comment]
Note - next major release will have RN0.60 as a peer dependency, so the instructions for <RN0.60 will be removed before then. I'm leaving both instructions in (for now), for users who are installing the library between now and the next major release, and are using the instructions from the master branch on GitHub.

**Comment by kadikraman** [comment]
Released in [v5.0.0](https://github.com/FormidableLabs/react-native-app-auth/releases/tag/v5.0.0) 🎉

**Pull request merged**